File: backend/core-services/product-service/src/main/java/com/movieticket/product/entity/Artist.java
package com.movieticket.product.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "artists")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Artist {
  @Id
  @GeneratedValue(strategy = GenerationType.UUID)
  private String id;

  private String fullName;

  @Column(columnDefinition = "TEXT")
  private String bio;

  private LocalDate dateOfBirth;
  private String nationality;
  private String avatarUrl;

  @Enumerated(EnumType.STRING)
  private ArtistType type;

  private LocalDateTime createdAt;
  private LocalDateTime updatedAt;

  @PrePersist
  protected void onCreate() {
    createdAt = LocalDateTime.now();
    updatedAt = LocalDateTime.now();
  }

  @PreUpdate
  protected void onUpdate() {
    updatedAt = LocalDateTime.now();
  }
}

File: frontend/web/src/types/artist.ts
export interface ArtistResDTO {
    id: string;
    fullName: string;
    bio: string;
    dateOfBirth: string;
    nationality: string;
    avatarUrl: string;
    type?: string;
}

export interface ArtistSummary {
    id: string;
    fullName: string;
}

export const ALL_ARTIST_TYPES = [
    { label: "Diễn viên", value: "ACTOR" },
    { label: "Đạo diễn", value: "DIRECTOR" }
];
